admin/metrics/client.go

```diff
@@ -98,3 +98,158 @@ func (c *Client) AutoscalerSlotsRecommendations(ctx context.Context, limit, offs
+}
+
+type Usage struct {
+	MetricName string  `json:"metric_name"`
+	Amount     float64 `json:"amount"`
+}
+
+func (c *Client) GetProjectUsageAvailability(ctx context.Context, projectIDs []string, startTime, endTime time.Time) ([]*ProjectUsageAvailability, error) {
+	// Create the URL for the request
+	var runtimeHost string
+
+	// In production, the REST and gRPC endpoints are the same, but in development, they're served on different ports.
+	if strings.Contains(c.RuntimeHost, "localhost") {
+		runtimeHost = "http://localhost:8081"
+	} else {
+		runtimeHost = c.RuntimeHost
+	}
+
+	uri, err := url.Parse(runtimeHost)
+	if err != nil {
+		return nil, err
+	}
+
+	uri.Path = path.Join("/v1/instances", c.InstanceID, "/api/project-usage-availability")
+	/*  sql api  -
+	  SELECT
+		project_id,
+		MIN(time) as min_time,
+		MAX(time) as max_time
+	  FROM rill-metrics
+	  WHERE
+		project_id IN ({{ .args.project_ids }}) and time >= '{{ .args.start_time }}' and time < '{{ .args.end_time }}'
+	  GROUP BY 1
+	*/
+
+	// Add URL query parameters
+	qry := uri.Query()
+	qry.Add("start_time", startTime.Format(time.RFC3339))
+	qry.Add("end_time", endTime.Format(time.RFC3339))
+	// create a comma separated string of projectIDs with single quotes around each projectID
+	qry.Add("project_ids", fmt.Sprintf("'%s'", strings.Join(projectIDs, "','")))
+
+	uri.RawQuery = qry.Encode()
+	apiURL := uri.String()
+
+	// Create a new HTTP request
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the access token in the request headers
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.AccessToken))
+
+	// Send the request
+	client := http.DefaultClient
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	// Check the response status code
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("request failed with status code %d", resp.StatusCode)
+	}
+
+	// Decode the JSON response into ProjectUsageAvailability structs
+	var availability []*ProjectUsageAvailability
+	err = json.NewDecoder(resp.Body).Decode(&availability)
+	if err != nil {
+		return nil, err
+	}
+
+	return availability, nil
+}
+
+func (c *Client) GetProjectUsageMetrics(ctx context.Context, projectID string, startTime, endTime time.Time, metricNames []string) ([]Usage, error) {
+	// Create the URL for the request
+	var runtimeHost string
+
+	// In production, the REST and gRPC endpoints are the same, but in development, they're served on different ports.
+	if strings.Contains(c.RuntimeHost, "localhost") {
+		runtimeHost = "http://localhost:8081"
+	} else {
+		runtimeHost = c.RuntimeHost
+	}
+
+	uri, err := url.Parse(runtimeHost)
+	if err != nil {
+		return nil, err
+	}
+
+	var usage []Usage
+	for _, metric := range metricNames {
+		// metric name is the api name
+		uri.Path = path.Join("/v1/instances", c.InstanceID, fmt.Sprintf("/api/%s", metric+"-usage"))
```

Comment:
Can we refactor to have a 1:1 mapping between function calls and network requests? Makes it easier to think about concurrency, and the loop can be easily handled at the caller